internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -602,6 +602,10 @@ func (a *Adapter) EnsureGroupSnapshotExist() (controller.OperationResult, error)
 
 	groupSnapshot, componentSnapshotInfos, err := a.prepareGroupSnapshot(a.application, prGroupHash)
 	if err != nil {
+		if strings.Contains(err.Error(), "failed to get pull request for GitHub") {
```

Comment:
I wonder is this would work for both Gitlab and Github , then we better make the error message  more informative as put what is the related git provider , wdyt?

Reply:
Yes, I think you are correct, though I believe there is a few other sections of code from[ group snapshots](https://issues.redhat.com/browse/STONEINTG-998) that refer to github that apply to gitlab as well.